src/main/java/com/socialfeed/domain/DataProvider.java

```diff
@@ -71,26 +72,64 @@
 		HashSet<String> ids = new HashSet<String>();
 		try
 		{
-			if (tableName.contains("Connection"))
+			String urlName = EndpointConstants.HOST + ":";
+			Gson g = new Gson();
+			List<Map<String, String>> idList = null;
+			URL url;
+			HttpURLConnection conn;
+			String data;
+			switch(tableName)
 			{
-				resultSet = sqlConnection.getTableRowByColumnName(tableName, "user1", id);
-				DataProvider.updateIdSet(resultSet, ids);
+			case DataProvider.CONNECTION_TABLE:
+				String getUserIds = urlName + EndpointConstants.CORE_PORT + "/" + EndpointConstants.USER_CONNECTION;
+				url = new URL(getUserIds + String.format("?user1={0}&type=Friend", id));
+				conn = (HttpURLConnection) url.openConnection();
+				
+				data = DataProvider.getJsonDataFromConnection(conn);
+				conn.disconnect();
 				
-				resultSet = sqlConnection.getTableRowByColumnName(tableName, "user2", id);
-				DataProvider.updateIdSet(resultSet, ids);
+				idList = g.fromJson(data, new TypeToken<List<Entity>>(){}.getType());
+				
+				url = new URL(getUserIds + String.format("?user2={0}", id));
+				conn = (HttpURLConnection) url.openConnection();
+				data = DataProvider.getJsonDataFromConnection(conn);
+				conn.disconnect();
+				idList.addAll(g.fromJson(data, new TypeToken<List<Map<String, String>>>(){}.getType()));
+				break;
+				
+			case DataProvider.EVENT_RELATION_TABLE:
+				String getEventIds = urlName + EndpointConstants.CORE_PORT + "/" + EndpointConstants.EVENT_ATTENDANCE_RELATION +
+					String.format("?user_id={0}&type=Accepted&type=Maybe", id);
+				url = new URL(getEventIds);
+				conn = (HttpURLConnection) url.openConnection();
 
-			}
-			else
-			{
-				resultSet = sqlConnection.getTableRowByColumnName(tableName, "user_id", id);
-				DataProvider.updateIdSet(resultSet, ids);
-			}
+				data = DataProvider.getJsonDataFromConnection(conn);
+				
+				conn.disconnect();
+				break;
 			
-			while (resultSet.next())
-			{
-				ids.add(resultSet.getString("id"));
+			case DataProvider.GROUP_MEMBERSHIP_TABLE:
+				String getGroupIds = urlName + EndpointConstants.CORE_PORT + "/" + EndpointConstants.GROUP_MEMBERSHIP +
+					String.format("?user_id={0}&type=Member", id);
+				url = new URL(getGroupIds);
+				conn = (HttpURLConnection) url.openConnection();
```

Comment:
can we refactor some of these common things?, like url = new.. conn , data, disconnect